roles/download/templates/kubeadm-images.yaml.j2

```diff
@@ -21,5 +21,5 @@ etcd:
 {% endif %}
 dns:
   type: CoreDNS
-  imageRepository: {{ coredns_image_repo | regex_replace('/coredns.*$','') }}
+  imageRepository: {{ coredns_image_repo | regex_replace('/coredns$','') }}
```

Comment:
The above has been updated since https://github.com/kubernetes-sigs/kubespray/pull/7570 for coredns versions
<=1.7.0.
It would be safer to have some condition for switching the above for versions.

Reply:
with https://github.com/kubernetes/kubernetes/commit/38a41e1557649a7cc763bf737779db9aa03ec75e and kubernetes release `1.21.2` kubeadm's constant `CoreDNSImageName` has been changed from `coredns/coredns` to `coredns`.

Because of that setting `imageRepository` to `k8s.gcr.io/coredns` to pull `k8s.gcr.io/coredns/coredns:v1.8.0` makes sense to me, but from what i see this should be handled automatically by `kubeadm` itself and this change should not be necessary

Reply:
with https://github.com/kubernetes/kubernetes/commit/38a41e1557649a7cc763bf737779db9aa03ec75e and kubernetes release `1.21.2` kubeadm's constant `CoreDNSImageName` has been changed from `coredns/coredns` to `coredns`.

Because of that setting `imageRepository` to `k8s.gcr.io/coredns` to pull `k8s.gcr.io/coredns/coredns:v1.8.0` makes sense to me, but from what i see this should be handled automatically by `kubeadm` itself and this change should not be necessary